common/status.py

```diff
@@ -0,0 +1,411 @@
+$XDG_STATE_HOME/backintime-backup-status.json
+(typically: ~/.local/state/backintime-backup-status.json).
+"""
+import json
+import os
+from pathlib import Path
+from datetime import datetime
+import config
+import logger
+import sshtools
+from exceptions import MountException
+import snapshots
+
+
+class BackupStatus:
+    """
+    Reports the status of the most recent backup for all profiles.
+
+    The status includes details about the last successful backup, the last
+    run, backup mode, and relevant paths. It can be formatted as a
+    human-readable string or JSON.
+
+    Args:
+        cfg: Configuration object.
+        all_status: Status for all profiles.
+        issues: If True, only profiles with issues are shown.
+        json: If True, output is in JSON format.
+    """
+
+    def __init__(self,
+                 cfg: config.Config,
+                 all_status: bool = True,
+                 issues: bool = False,
+                 format_json: bool = False):
+
+        self.cfg = cfg
+        self.issues = issues
+        self.json = format_json
+        self.all_status = all_status
+        self.status = {}
+
+    def get_status(self):
+        """
+        Get the formatted status of the backup profiles.
+
+        Returns:
+            str:
+                        Human-readable or JSON-formatted string (depending
+                        on self.json).
+        """
+        self._read_status_file()
+
+        status = self.status.get(
+            self.cfg.profileName(),
+            None
+        )
+
+        if status is None or status['Last Run'] in [None, 'Unknown']:
+
+            logger.warning(
+                f'No status found for profile "{self.cfg.currentProfile()}". '
+                'Trying to create new status entry.'
+            )
+            self.update_status()
+
+        return self._get_formatted_status()
+
+    def update_status(self, timestamp: datetime = None):
+        """
+        Update the status for the current profile, called after a backup
+        attempt.
+
+        If no status file exists, all profiles will be checked and
+        the status file will be created. The selected profile will then
+        be updated to ensure it is using the timestamp of the last
+        backup attempt.
+
+        Args:
+            timestamp (datetime, optional):
+                        Timestamp to use as the last run time.
+        """
+        profile_name = self.cfg.profileName()
+        self._read_status_file()
+        self.status[profile_name] = self._create_profile_status(timestamp)
+        self._write_status_file()
+
+    def _read_status_file(self):
+        """
+        Read the backup status file from disk and store the data in
+        self.status. If the file does not exist or is invalid, create a new
+        status file for all profiles.
+        """
+        try:
+            with open(_status_file_path(), 'r', encoding='utf-8') as f:
+                self.status = json.load(f)
```

Comment:
One letter variable names are not a good choice.

```suggestion
            with open(_status_file_path(), 'r', encoding='utf-8') as handle:
                self.status = json.load(handle)
```